lib/vanagon/project.rb

```diff
@@ -476,5 +476,32 @@ def generate_package
     def generate_packaging_artifacts(workdir)
       @platform.generate_packaging_artifacts(workdir, @name, binding)
     end
+
+    # Generate a json hash which lists all of the dependant components of the
+    # project.
+    #
+    # @return [Hash] where the top level keys are components and their values
+    #   are hashes with additional information on the component.
+    def generate_dependencies_info
+      @components.each_with_object({}) do |component, hsh|
+        hsh.merge(component.get_dependency_hash)
```

Comment:
You tested that `#merge` will do the right thing, and that you don't need `#merge!`?